or_issue.py

```diff
@@ -0,0 +1,210 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import sys
+import shutil
+import pathlib
+import argparse
+from collections import deque
+from os.path import join, abspath, dirname, basename, isdir, relpath
+
+openlane_path = abspath(dirname(__file__))
+
+parser = argparse.ArgumentParser(description="""
+This script creates a reproducible, self-contained package of files to demonstrate
+OpenROAD behavior in a vaccum, suitable for filing issues.
+
+The result is a tarred, xzipped and gzipped tarball for minimal size while
+maintaining GitHub uploadability.
+
+Requires UNIX-like operating system. Final package printed to /dev/stdout. Don't forget to chomp/rstrip.
+
+Usage example: IMAGE_NAME=efabless/openlane:v0.15 python3 or_issue.py -s ./scripts/openroad/or_pdn.tcl ./designs/inverter/runs/openlane_test
+""")
+parser.add_argument('--or-script', '-s', required=True, help='Name of the OpenROAD script causing the failure: i.e. or_antenna_check.tcl, or_pdn.tcl, etc.')
+parser.add_argument('--pdk-root', required=(os.getenv("PDK_ROOT") is None), default=os.getenv("PDK_ROOT"), help='Path to the PDK root (Required if environment variable PDK_ROOT is not set.)')
+parser.add_argument('run_path', help='Path to the run folder.')
+parser.add_argument('--no-tarball', action="store_true", help="Do not create a tarball.")
+args = parser.parse_args()
+
+script_path = abspath(args.or_script)
+run_path = abspath(args.run_path)
+pdk_root = abspath(args.pdk_root)
+tarball = not args.no_tarball
+or_scripts_path = join(openlane_path, "scripts", "openroad")
+
+if not script_path.startswith(or_scripts_path):
+    print(f"⚠ The OpenROAD script {script_path} does not appear to be in {or_scripts_path}.", file=sys.stderr)
+    print("This script's behavior may be undefined.", file=sys.stderr)
+
+run_name = basename(run_path)
+script_basename = basename(args.or_script)[:-4]
+
+script_path_containerized = script_path.replace(openlane_path, "/openLANE_flow")
+run_path_containerized = run_path.replace(openlane_path, "/openLANE_flow")
+
+
+# Phase 1: Read All Environment Variables
+# pdk_config = join(args.pdk_root, "sky130A", "libs.tech", "openlane", "config.tcl")
+print(f"Parsing config file(s)...", file=sys.stderr)
+run_config = join(run_path, "config.tcl")
+
+env = {}
+
+
+def read_env(config_path: str, from_path: str, input_env={}) -> dict:
+    rx = r"\s*set\s*::env\((.+?)\)\s*(.+)"
+    env = input_env.copy()
+    string_data = ""
+    try:
+        string_data = open(config_path).read()
+    except FileNotFoundError:
+        print(f"❌ File {config_path} not found. The {from_path} path may have been specified incorrectly.", file=sys.stderr)
+        exit(66)
```

Comment:
What does 66 denote?

Reply:
EX_NODATA, a [BSD exit code](https://www.freebsd.org/cgi/man.cgi?query=sysexits&apropos=0&sektion=0&manpath=FreeBSD+4.3-RELEASE&format=html). It's probably the most ignored standard in existence at this point but I just support it where I can.

Reply:
Use os.EX_NOINPUT in that case (there is no EX_NODATA afaik)

Reply:
Yeah, sorry, that was a pre-coffee message. I didn't know python enumerated sysexits as well, that's kinda cool